app/plugin_api.go

```diff
@@ -759,6 +760,13 @@ func (api *PluginAPI) UploadFile(data []byte, channelID string, filename string)
 	return api.app.UploadFile(api.ctx, data, channelID, filename)
 }
 
+func (api *PluginAPI) UploadFileFromReader(reader io.Reader, channelID string, filename string) (*model.FileInfo, *model.AppError) {
+	return api.app.UploadFileX(api.ctx, channelID, filename, reader,
+		UploadFileSetUserId("nouser"),
```

Comment:
Passing "nouser" seems incorrect. I'd suggest to perhaps make `userID` a param itself. Alternatively, there's also `api.ctx.Session().UserId` but that seems to be initialized at startup, so likely will be empty.

Reply:
@agnivade My intention was to mimic the implementation of the [UploadFile](https://developers.mattermost.com/integrate/plugins/server/reference/#API.UploadFile) plugin API method, which does not require a user ID. Most methods of the plugin API do not require a user ID, unless the operation absolutely requires the user ID.

I'm open to the idea of requiring the user ID here, though I think this sort of changes the general patterns of the plugin API methods, and we should rethink this with how we want to continue adding new methods. cc @hanzei

Reply:
ok